components/ui-api-layer/internal/domain/k8s/resourcequota_resolver.go

```diff
@@ -30,3 +71,153 @@ func (r *resourceQuotaResolver) ResourceQuotasQuery(ctx context.Context, environ
 
 	return r.converter.ToGQLs(items), nil
 }
+
+func (r *resourceQuotaResolver) ResourceQuotaStatus(ctx context.Context, environment string) (gqlschema.ResourceQuotaStatus, error) {
+	const (
+		rsKind     = "ReplicaSet"
+		ssKind     = "StatefulSet"
+		deployKind = "Deployment"
+	)
+
+	resourcesToCheck := []v1.ResourceName{
+		v1.ResourceRequestsMemory,
+		v1.ResourceLimitsMemory,
+		v1.ResourceRequestsCPU,
+		v1.ResourceLimitsCPU,
+		v1.ResourcePods,
+	}
+
+	resourceQuotas, err := r.rqLister.ListResourceQuotas(environment)
+	if err != nil {
+		return gqlschema.ResourceQuotaStatus{}, errors.Wrapf(err, "while listing ResourceQuotas [environment: %s]", environment)
+	}
+	for _, rq := range resourceQuotas {
+		for _, name := range resourcesToCheck {
+			hard, hardExists := rq.Spec.Hard[name]
+			used := rq.Status.Used[name]
+			if hardExists && used.Value() >= hard.Value() {
+				return gqlschema.ResourceQuotaStatus{
+					Exceeded: true,
+					Message:  fmt.Sprintf("Your ResourceQuota `%s` limit exceeded", name),
+				}, nil
+			}
+		}
+	}
+
+	replicaSets, err := r.rsLister.ListReplicaSets(environment)
+	if err != nil {
+		return gqlschema.ResourceQuotaStatus{}, errors.Wrapf(err, "while listing ReplicaSets [environment: %s]", environment)
+	}
+	for _, rs := range replicaSets {
+		var maxUnavailable int32
+		if len(rs.GetOwnerReferences()) > 0 {
+			for _, ownerRef := range rs.GetOwnerReferences() {
+				if ownerRef.Kind == deployKind {
+					maxUnavailable, err = r.extractMaxUnavailable(environment, ownerRef.Name, *rs.Spec.Replicas)
+					if err != nil {
+						return gqlschema.ResourceQuotaStatus{}, err
+					}
+					break
+				}
+			}
+		}
+		if *rs.Spec.Replicas > rs.Status.Replicas+maxUnavailable {
+			status, err := r.checkPodsUsage(environment, rs.Spec.Selector.MatchLabels, resourcesToCheck, resourceQuotas)
```

Comment:
a name 'status' does not describe the variable, maybe 'exceeded'? the same for stateful sets